// openregister/resources/company.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { path } from "../internal/utils/path.mjs";
export class Company extends APIResource {
    /**
     * Get company contact information
     */
    getContactV0(companyID, options) {
        return this._client.get(path `/v0/company/${companyID}/contact`, options);
    }
    /**
     * Get detailed company information
     */
    getDetailsV1(companyID, query = {}, options) {
        return this._client.get(path `/v1/company/${companyID}`, { query, ...options });
    }
    /**
     * Get financial reports
     */
    getFinancialsV1(companyID, options) {
        return this._client.get(path `/v1/company/${companyID}/financials`, options);
    }
    /**
     * Get company holdings
     */
    getHoldingsV1(companyID, options) {
        return this._client.get(path `/v1/company/${companyID}/holdings`, options);
    }
    /**
     * Get company owners
     */
    getOwnersV1(companyID, query = {}, options) {
        return this._client.get(path `/v1/company/${companyID}/owners`, { query, ...options });
    }
    /**
     * Get company end owners
     */
    getUbosV1(companyID, options) {
        return this._client.get(path `/v1/company/${companyID}/ubo`, options);
    }
}
