// openregister/resources/company.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Company = void 0;
const resource_1 = require("../core/resource.js");
const path_1 = require("../internal/utils/path.js");
class Company extends resource_1.APIResource {
    /**
     * Get company contact information
     */
    getContactV0(companyID, options) {
        return this._client.get((0, path_1.path) `/v0/company/${companyID}/contact`, options);
    }
    /**
     * Get detailed company information
     */
    getDetailsV1(companyID, query = {}, options) {
        return this._client.get((0, path_1.path) `/v1/company/${companyID}`, { query, ...options });
    }
    /**
     * Get financial reports
     */
    getFinancialsV1(companyID, options) {
        return this._client.get((0, path_1.path) `/v1/company/${companyID}/financials`, options);
    }
    /**
     * Get company holdings
     */
    getHoldingsV1(companyID, options) {
        return this._client.get((0, path_1.path) `/v1/company/${companyID}/holdings`, options);
    }
    /**
     * Get company owners
     */
    getOwnersV1(companyID, query = {}, options) {
        return this._client.get((0, path_1.path) `/v1/company/${companyID}/owners`, { query, ...options });
    }
    /**
     * Get company end owners
     */
    getUbosV1(companyID, options) {
        return this._client.get((0, path_1.path) `/v1/company/${companyID}/ubo`, options);
    }
}
exports.Company = Company;
